core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -85,8 +85,16 @@ public void run(final ActionContext context) throws Exception {
 
       LOG.debug("Output: {}", out);
       LOG.error("Errors: {}", err);
+      Integer exitCode = session.getExitStatus();
+      if (exitCode != null && exitCode > 0) {
+        throw new IOException(String.format("Error: command finished with exit code: %d", exitCode));
```

Comment:
lets put the actual command and hostname in the error message. Otherwise somebody sees the log, but then has to figure out which pipeline and action its from in order to find the actual command.